### ISSUE TITLE
Performance/InefficientHashSearch false positive on non-hash

### ISSUE THREAD

**Issue by ian-h-davidson** [rubocop-hq/rubocop#6215]
The cop `Performance/InefficientHashSearch` makes the assumption that anything that responds to `.keys.include?` also responds to `key?`, which is not always the case. 

This cop was first purposed awhile ago in #3707, where this issue was also mentioned as a possibility.

--------

## Expected behavior

`#.keys.include?` should be acceptable when on non hashes.

## Actual behavior

`#.keys.include?` is suggested to be replaced with `#.keys?`

## Steps to reproduce the problem
The following will be deemed bad:
```
class Test
  attr_accessor :keys
  def initialize
    @keys = []
  end
end

t = Test.new
puts t.keys.include? 'key'
```

`keys.include?` is suggested to be replaced by `key?`

## RuboCop version

```
$ rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.3.4 x86_64-darwin16)
```

**Comment by rrosenblum** [comment]
This is a limitation of static code analysis. There is no way to fix this.

**Comment by ian-h-davidson** [comment]
Ah, that's a shame.

**Comment by MatzFan** [comment]
FWIW it appears the same is true of `values.include?`. Iterating a `Nokogiri::XML::Node` gives a false positive, but the class does not implement a `#value?` method. I may suggest the maintainers add `#key?` and `#value?` via a PR, as it would appear good practice for hash-like objects.

**Comment by ngouy** [comment]
or with Structs
```rb
Struct.new(:a, :b).new('a', 'b').values
# => ["a", "b"]
Struct.new(:a, :b).new('a', 'b').value?("b")
# NoMethodError: undefined method `value?' for #<struct a="a", b="b">
# Did you mean?  values
```
